ml-agents/tests/yamato/yamato_utils.py

```diff
@@ -49,21 +49,62 @@ def run_standalone_build(base_path: str, verbose: bool = False) -> int:
     return res.returncode
 
 
-def init_venv():
+def init_venv(mlagents_python_version: str = None) -> str:
+    """
+    Set up the virtual environment, and return the venv path.
+    :param mlagents_python_version: The version of mlagents python packcage to install.
+        If None, will do a local install, otherwise will install from pypi
+    :return:
+    """
+    # Use a different venv path for different versions
+    venv_path = "venv"
+    if mlagents_python_version:
+        venv_path += "_" + mlagents_python_version
+
     # Set up the venv and install mlagents
-    subprocess.check_call("python -m venv venv", shell=True)
+    subprocess.check_call(f"python -m venv {venv_path}", shell=True)
     pip_commands = [
         "--upgrade pip",
         "--upgrade setuptools",
         # TODO build these and publish to internal pypi
         "~/tensorflow_pkg/tensorflow-2.0.0-cp37-cp37m-macosx_10_14_x86_64.whl",
-        "-e ./ml-agents-envs",
-        "-e ./ml-agents",
     ]
+    if mlagents_python_version:
+        # install from pypi
+        pip_commands.append(f"mlagents=={mlagents_python_version}")
+    else:
+        # Local install
+        pip_commands += ["-e ./ml-agents-envs", "-e ./ml-agents"]
     for cmd in pip_commands:
         subprocess.check_call(
-            f"source venv/bin/activate; python -m pip install -q {cmd}", shell=True
+            f"source {venv_path}/bin/activate; python -m pip install -q {cmd}",
+            shell=True,
         )
+    return venv_path
+
+
+def checkout_csharp_version(csharp_version):
+    """
+    Checks out the specific git revision (usually a tag) for the C# package and Project.
+    If csharp_version is None, no changes are made.
+    :param csharp_version:
+    :return:
+    """
+    if csharp_version is None:
+        return
+    csharp_dirs = ["com.unity.ml-agents", "Project"]
+    for csharp_dir in csharp_dirs:
+        subprocess.check_call(
+            f"git checkout {csharp_version} -- {csharp_dir}", shell=True
+        )
+
+
+def undo_git_checkout():
+    """
+    Clean up the git working directory.
+    """
+    subprocess.check_call("git reset HEAD .", shell=True)
```

Comment:
Number of times I accidentally reverted changes to this file while testing these commands: 2

Reply:
I created a git utility class while working at nvidia in order to be able to mock out the git calls for testing.  saved my life.

Reply:
YAGNI for now, but it if we need to do more git logic, I'll look into it.